lib/autofill.js

```diff
@@ -2,93 +2,118 @@
+        .then(radios => {
+          debug(`Checking random radio: ${name}`);
+          const index = 1 + Math.floor(Math.random() * (radios.value.length - 1));
+          return browser.elementIdClick(radios.value[index].ELEMENT);
+        });
+    }
+    return browser.elementIdAttribute(element, 'value')
+      .then(val => {
+        if (val.value === value) {
+          debug(`Checking radio: ${name} with value: ${val.value}`);
+          browser.elementIdClick(element);
+        }
+      });
+  }
+
+  function completeCheckbox(element, name) {
+    const value = getValue(name, 'checkbox');
     return browser.elementIdAttribute(element, 'value')
       .then(val => {
         return browser.elementIdAttribute(element, 'checked')
           .then((checked) => {
-            console.log(checked.value);
-            if (val.value === value && !checked.value) {
-              debug(`Checking ${type}: ${name} with value: ${val.value}`);
+            if (!value && !checked.value) {
+              debug(`Checking checkbox: ${name} with value: ${val.value}`);
+              browser.elementIdClick(element);
+            } else if (value.includes(val.value) && !checked.value) {
+              debug(`Checking checkbox: ${name} with value: ${val.value}`);
               browser.elementIdClick(element);
-            } else if (val.value !== value && checked.value) {
-              debug(`Unchecking ${type}: ${name} with value: ${val.value}`);
+            } else if (!value.includes(val.value) && checked.value) {
+              debug(`Unchecking checkbox: ${name} with value: ${val.value}`);
               browser.elementIdClick(element);
             } else {
-              debug(`Ignoring ${type}: ${name} with value: ${val.value} - looking for ${value}`);
+              debug(`Ignoring checkbox: ${name} with value: ${val.value} - looking for ${value}`);
             }
           });
       });
   }
 
   function completeSelectElement(element, name) {
-    const value = getValue(name);
-    debug(`Selecting options: ${value} from select box: ${name}`);
-    browser.selectByValue(`select[name="${name}"]`, value);
+    const value = getValue(name, 'select');
+    if (!value) {
+      return browser.elementIdElements(element, 'option')
+        .then((options) => {
+          const index = 1 + Math.floor(Math.random() * (options.value.length - 1));
+          debug(`Selecting option: ${index} from select box: ${name}`);
+          return browser.selectByIndex(`select[name="${name}"]`, index);
+        });
+    } else {
+      debug(`Selecting options: ${value} from select box: ${name}`);
+      return browser.selectByValue(`select[name="${name}"]`, value);
+    }
   }
 
   function completeStep(path) {
     return browser
       .elements('input')
       .then(fields => {
         debug(`Found ${fields.value.length} <input> elements`);
-        const fillers = fields.value.map(field => {
+        return Promise.map(fields.value, field => {
```

Comment:
what is this trickery! `Promise.map` isn't a 'thing' with native promises... hmm... Bluebird eh?